client/extensions/woocommerce/state/selectors/promotions.js

```diff
@@ -58,3 +58,8 @@ export function getPromotionWithLocalEdits( rootState, promotionId, siteId = get
 	return null;
 }
 
+export function getPromotionableProducts( rootState, siteId = getSelectedSiteWithFallback( rootState ) ) {
+	const promotions = get( rootState, [ 'extensions', 'woocommerce', 'sites', siteId, 'promotions' ], {} );
+	return promotions.products;
```

Comment:
Should this return an empty array if `promotions.products` doesn't exist?

Reply:
I prefer to leave it as undefined so I can tell if it's been loaded or not.